godot-core/src/builtin/mod.rs

```diff
@@ -128,3 +128,205 @@ pub(crate) fn u8_to_bool(u: u8) -> bool {
         _ => panic!("Invalid boolean value {u}"),
     }
 }
+
+/// Clippy often complains if you do `f as f64` when `f` is already an `f64`.
+/// This trait exists to make it easy to convert between the different reals
+/// and floats without a lot of allowing clippy lints for your code.
+pub trait RealConv {
+    /// Cast this [`real`] to an [`f32`] using `as`.
+    // Clippy complains that this is an `as_*` function but it takes a `self`
+    // however, since this uses `as` internally it makes much more sense for
+    // it to be named `as_f32` rather than `to_f32`.
+    #[allow(clippy::wrong_self_convention)]
+    fn as_f32(self) -> f32;
```

Comment:
In Rust, `as_` functions almost always return a reference or raw pointer by convention. 
They express: "return a different representation of _this very_ object", i.e. similar to a transmute, and not a copy.

On the other hand, `to_` suggests that a different type is returned (a conversion takes place), but here this is only true in half the cases. So maybe not ideal either.

I'm not very sure about this, more opinions on this matter?

Reply:
it does look more similar to the code the user wants to write, i.e
```rs
r as f64
```
vs
```rs
r.as_f64()
```